examples/HK.jl

```diff
@@ -0,0 +1,112 @@
+# In a Synchronous update schedule changes made to an agent are not seen by
+# other agents until the next clock tick — that is,
+# all agents update simultaneously [Wilensky 2015, p.286](https://mitpress.mit.edu/books/introduction-agent-based-modeling)
+
+
+# The model has the following components:
+
+# - A set of n Agents with opinions xᵢ in the range [0,1] as attribute;
+# - A bound ϵ in also in the range [0,1] (actually, the range of interesting results is
+# approximately (0, 0.3]);
+# - The update rule: at each step every agent adopts the mean of the opinions which are within
+# the confidence bound ( |xᵢ - xⱼ| ≤ ϵ);
+
+
+# ## Core structures
+# We start by defining the Agent type and initializing the model.
+# The Agent type has two fields so that we can implement the synchronous update.
+using Agents
+using Distributions: mean
+
+
+mutable struct HKAgent{T <: AbstractFloat} <: AbstractAgent
+    id::Int
+    old_opinion::T
+    new_opinion::T
+end
+
+
+function hk_model(;numagents = 100, ϵ = 0.4)
+    model = ABM(HKAgent, scheduler = fastest,
+                properties = Dict(:ϵ => ϵ))
+    for i in 1:numagents
+        o = rand()
+        add_agent!(model, o, o)
+    end
+    return model
+end
+
+# And some helper functions for the update rule. As there is a filter in
+#the rule we implement it outside the `agent_step!` method. Notice that the filter
+# is applied to the  `:old_opinion` field .
+get_old_opinion(agent)::Float64 = agent.old_opinion
+
+function boundfilter(agent,model)
+    filter(j->abs(get_old_opinion(agent) - j) < model.properties[:ϵ],
+     get_old_opinion.(values(model.agents)))
+end
+
+# Now we implement the `agent_step!` and `model_step!` methods.
+function agent_step!(agent, model)
+    agent.new_opinion = mean(boundfilter(agent,model))
+end
+
+function model_step!(model)
+    for i in keys(model.agents)
+        agent = id2agent(i, model)
+        agent.old_opinion = agent.new_opinion
+    end
+end
+
+# From this implementation we see that to implement synchronous scheduling
+# we define an Agent type with an `old` and `new` fields for attributes that
+# are changed through synchronous updating. In the agent_step! we use the `old` field
+# and after updating all the agents `new` field we use the `model_step!`
+# to update the model  for the next iteration.
+
+# Now we can define a method for our simulation run.
+# The parameter of interest is the :new_opinion` field so we assign
+# it to variable agent_properties and pass it to the `step!` method
+# to be collected in a DataFrame.
+function model_run(; numagents = 100, iterations = 50, ϵ= 0.3)
+    model = hk_model(numagents = numagents, ϵ = ϵ)
+    when = 0:5:iterations
+    agent_properties = [:new_opinion]
+    data = step!(
+            model,
+            agent_step!,
+            model_step!,
+            iterations,
+            agent_properties,
+            when = when
+            )
+    return(data)
+end
```

Comment:
Here I think there should be an output of `model_run`

Reply:
I'm going to add that and try to run the docs locally